### PR TITLE
Fix to GroupReadsByUmi to ensure that all reads with same start/end a…

### PR DESCRIPTION
**Pull request by tfenne**
…re seen together.

### CODE REVIEW

File: src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala [hunk 1]
```diff
@@ -50,9 +52,100 @@ object GroupReadsByUmi {
   /** A type to represent a unique ID assigned to a molecule. */
   type MoleculeId = String
 
-  private val SortPositionAttribute        = "__GRs"
-  private val LibraryAttribute             = "__GRlb"
-  private val InsertEndsAndStrandAttribute = "__Insert"
+  private val ReadEndsTempAttributeName = "__GRBU_ReadInfo"
+
+  /** A case class to represent all the information we need to order reads for duplicate marking / grouping. */
+  case class ReadInfo(refIndex:Int, start1:Int, start2: Int, strand1: Boolean, strand2: Boolean, library:String) extends Ordered[ReadInfo] {
```
Comment (nh13) [inline src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala]: spaces after colons?

File: src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala [hunk 2]
```diff
@@ -50,9 +52,100 @@ object GroupReadsByUmi {
   /** A type to represent a unique ID assigned to a molecule. */
   type MoleculeId = String
 
-  private val SortPositionAttribute        = "__GRs"
-  private val LibraryAttribute             = "__GRlb"
-  private val InsertEndsAndStrandAttribute = "__Insert"
+  private val ReadEndsTempAttributeName = "__GRBU_ReadInfo"
```
Comment (nh13) [inline src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala]: `ReadInfoTempAttributeName`?

File: src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala [hunk 3]
```diff
@@ -254,84 +347,9 @@ object GroupReadsByUmi {
...
-        val pick = Math.min(p1, p2)
-        rec.setTransientAttribute(SortPositionAttribute, pick)
-        pick
-      }
-    }
-  }
 }
 
-@clp(group=ClpGroups.Umi, description =
+@sopt.clp(group=ClpGroups.Umi, description =
```
Comment (nh13) [inline src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala]: ?

**Comment by nh13** [comment]
Minor comments, and tests are failing.

File: src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala (hunk 3 above)
Comment (tfenne) [inline src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala]: Stupid intellij.  Will fix.

**Pull request merged**